File: LeetCode/src/main/java/com/jack/linkedlist/LeetCode19.py

from typing import Optional

from ListNode import ListNode


class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        pre = ListNode(next=head)
        slow = fast = pre
        
        while n>0:
            fast = fast.next
            n-=1
            
        while fast.next:
            fast = fast.next
            slow = slow.next
            
        slow.next = slow.next.next

        return pre.next

if __name__ == "__main__":
    pass

File: LeetCode/src/main/java/com/jack/linkedlist/LeetCode19.java
package com.jack.linkedlist;

public class LeetCode19 {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode pre = new ListNode(0, head);
            ListNode fast = pre;
            ListNode slow = pre;
            int k = n;
            while(k>0) {
                fast = fast.next;
                k--;
            }
            while(fast.next != null){
                fast= fast.next;
                slow = slow.next;
            }

            slow.next= slow.next.next;

        return pre.next;
    
    }

}
